lib/manager/nuget/extract.ts

```diff
@@ -40,6 +78,7 @@ export function extractPackageFile(
         currentValue,
         managerData: { lineNumber },
         datasource: datasourceNuget.id,
+        registryUrls,
```

Comment:
With this change all NuGet deps will now have registry URLs. I'm a bit unsure about the effect of this - specifically whether and how this influences registry URLs from configuration and whether/how those URLs are passed to the datasource later on. Do I need to do anything or will it "just work" :)

Reply:
Ideally we'd leave `registryUrls` empty at the extract stage if they're unspecified and then add in a default registry at the datasource stage, but it doesn't *break* if you need to add them at the extract stage

Reply:
BTW this is relevant: https://github.com/renovatebot/renovate/pull/5686

Reply:
I would prefer to do it at config parsing stage.

So find all relevant `nuget.config` and add relevant config items as `packageRules`.

@rarkin Maybe we can add multiple config parser like managers?

Reply:
Aren't there two locations for Nuget.config? `%appdata%\NuGet\NuGet.Config` which in theory could be read during global init, and then a `NuGet.Config` per-repo? Not sure if there's even the possibility of one config file per subdirectory?

If the registryUrls are in the NuGet.config, then we should extract them. But when it comes to inserting default registryUrls, I prefer to do it in the datasource if possible.

What we're trying to do is still allow users to override *default* settings easily.

Reply:
Ok

https://docs.microsoft.com/en-us/nuget/consume-packages/configuring-nuget-behavior#config-file-locations-and-uses

Reply:
BTW I don't *really* like Renovate searching through folders on local machines in search of config files (apart from our own!) but we started down that path and now users expect it. I've already seen cases of people surprised that Renovate doesn't find its `nuget.config` on the VM that it runs on. I think we can add support for that at a later stage though - @fgreinacher I assume this is just for `nuget.config` within the repo?

Reply:
> @fgreinacher I assume this is just for nuget.config within the repo?

Yes, these changes are just for the straight-forward (and 95%) case where there is a single NuGet.config in the repo. 

> If the registryUrls are in the NuGet.config, then we should extract them. But when it comes to inserting default registryUrls, I prefer to do it in the datasource if possible.

We still need to handle the default registry URLs during extraction because a local NuGet.config always adds feeds or clears the existing feeds (see the new test fixture).

Reply:
> BTW I don't really like Renovate searching through folders on local machines in search of config files (apart from our own!) but we started down that path and now users expect it

Maybe we should stop the search at the repository root?

Reply:
Yes, and add a note to the docu, that we don't search nuget system paths?

Reply:
> BTW this is relevant: #5686

Cool, thanks! I'll wait for this to be merged, OK?

Reply:
> es, and add a note to the docu, that we don't search nuget system paths?

Done with cc9ab8fbfef571f0f3bd77bcf2043a902f517812